src/drivers/actuators/vertiq_io/vertiq_client_manager.hpp

```diff
@@ -0,0 +1,275 @@
+	*/
+	uint8_t GetNumberOfOperationalClients();
+
+	/**
+	* @brief Returns a pointer to our IFCI client used for telemetry
+	*
+	* @return The _telem_ifci pointer
+	*/
+	IQUartFlightControllerInterfaceClient *GetTelemIFCI();
+
+	/**
+	* @brief Creates and adds a new entry wrapper object to our array of entry wrappers
+	*
+	* @param px4_param A parameter stored in PX4. This can be found with the param_find function
+	* @param entry A pointer to a Vertiq client entry
+	*/
+	template <typename iquart_data_type , typename px4_data_type>
+	void AddNewClientEntry(param_t px4_param, ClientEntryAbstract *entry){
+		if(_added_entry_wrappers < MAX_CLIENT_ENTRIES){
+			_entry_wrappers[_added_entry_wrappers] = new EntryWrapper<iquart_data_type, px4_data_type>;
+			_entry_wrappers[_added_entry_wrappers]->ConfigureStruct(px4_param, entry);
+			_added_entry_wrappers++;
+		}else{
+			PX4_INFO("Could not add this entry. Maximum number exceeded");
+		}
+	}
+
+private:
+	/**
+	* @brief Initialize all of the Vertiq Clients that we want to use
+	*
+	* @param object_id The object ID with which to initialize our clients
+	*/
+	void InitVertiqClients(uint8_t object_id);
+
+	/**
+	* @brief Initialize all of the Entry Wrappers
+	*/
+	void InitEntryWrappers();
+
+	uint8_t _object_id_now;
+
+	//We need a serial handler in order to talk over the serial port
+	VertiqSerialInterface *_serial_interface;
+
+	//These exist whenever we're using IQUART
+	EntryWrapper<float, float> _velocity_max_entry;
+	EntryWrapper<float, float> _voltage_max_entry;
+	EntryWrapper<uint8_t, int32_t> _control_mode_entry;
+	EntryWrapper<uint8_t, int32_t> _motor_direction_entry;
+	EntryWrapper<uint8_t, int32_t> _fc_direction_entry;
+
+#ifdef CONFIG_USE_IFCI_CONFIGURATION
+	EntryWrapper<uint8_t, int32_t> _throttle_cvi_entry;
+#endif //CONFIG_USE_IFCI_CONFIGURATION
+
+#ifdef CONFIG_USE_PULSING_CONFIGURATION
+	EntryWrapper<uint8_t, int32_t> _pulsing_voltage_mode_entry;
+	EntryWrapper<uint8_t, int32_t> _x_cvi_entry;
+	EntryWrapper<uint8_t, int32_t> _y_cvi_entry;
+	EntryWrapper<float, float> _pulse_zero_angle_entry;
+	EntryWrapper<float, float> _pulse_velo_cutoff_entry;
+	EntryWrapper<float, float> _pulse_torque_offset_entry;
+	EntryWrapper<float, float> _pulse_volt_limit_entry;
+#endif //CONFIG_USE_PULSING_CONFIGURATION
+
+////////////////////////////////////////////////////////////////////////
+//Vertiq client information
+	//Some constants to help us out
+	static const uint8_t MAXIMUM_CONFIGURATION_CLIENTS = 20; //These are clients whose module ID will change when Target Module ID changes
+	static const uint8_t MAXIMUM_OPERATIONAL_CLIENTS = 20; //These are clients that are used for module control/telemetry. They have a static Module ID
+
+	//Client Arrays
+	ClientAbstract *_configuration_client_array[MAXIMUM_CONFIGURATION_CLIENTS];
+	uint8_t _configuration_clients_in_use = 0;
+
+	ClientAbstract *_operational_client_array[MAXIMUM_OPERATIONAL_CLIENTS];
+	uint8_t _operational_clients_in_use = 0;
```

Comment:
Similar comment as above, may be worthwhile expanding the commenting here to clarify what the different intentions of these different types of clients are so people can more easily understand how to use this